packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Permissions/PermissionsCard/ApplicationScopesAssignmentModal/index.tsx

```diff
@@ -0,0 +1,160 @@
+import { useTranslation } from 'react-i18next';
+
+import ConfirmModal from '@/ds-components/ConfirmModal';
+import DataTransferBox from '@/ds-components/DataTransferBox';
+import TabNav, { TabNavItem } from '@/ds-components/TabNav';
+import TabWrapper from '@/ds-components/TabWrapper';
+
+import {
+  allLevelPermissionTabs,
+  organizationLevelPermissionsTab,
+  userLevelPermissionsTabs,
+} from './constants';
+import { ScopeLevel } from './type';
+import useApplicationScopesAssignment from './use-application-scopes-assignment';
+
+type Props = {
+  readonly isOpen: boolean;
+  readonly onClose: () => void;
+  readonly applicationId: string;
+  readonly scopeLevel: ScopeLevel;
+};
+
+function ApplicationScopesAssignmentModal({ isOpen, onClose, applicationId, scopeLevel }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+
+  const { activeTab, setActiveTab, scopesAssignment, clearSelectedData, onSubmit, isLoading } =
+    useApplicationScopesAssignment(applicationId, scopeLevel);
+
+  const onCloseHandler = useCallback(() => {
+    onClose();
+    clearSelectedData();
+    setActiveTab(
+      scopeLevel === ScopeLevel.Organization
+        ? ApplicationUserConsentScopeType.OrganizationResourceScopes
+        : ApplicationUserConsentScopeType.UserScopes
+    );
+  }, [clearSelectedData, onClose, scopeLevel, setActiveTab]);
+
+  const onSubmitHandler = useCallback(async () => {
+    await onSubmit();
+    onCloseHandler();
+  }, [onCloseHandler, onSubmit]);
+
+  // If any of the tabs has selected scopes, the modal is dirty
+  const isDirty = useMemo(
+    () => Object.values(scopesAssignment).some(({ selectedData }) => selectedData.length > 0),
+    [scopesAssignment]
+  );
+
+  const tabs = useMemo(() => {
+    const getPermissionTabs = () => {
+      if (scopeLevel === ScopeLevel.All) {
+        return allLevelPermissionTabs;
+      }
+
+      return scopeLevel === ScopeLevel.User
+        ? userLevelPermissionsTabs
+        : organizationLevelPermissionsTab;
+    };
+
+    return Object.values(getPermissionTabs()).map(({ title, key }) => {
+      const selectedDataCount = scopesAssignment[key].selectedData.length;
+
+      return (
+        <TabNavItem
+          key={key}
+          isActive={key === activeTab}
+          onClick={() => {
+            setActiveTab(key);
+          }}
+        >
+          {`${String(t(title))}${selectedDataCount ? ` (${selectedDataCount})` : ''}`}
+        </TabNavItem>
+      );
+    });
+  }, [activeTab, scopeLevel, scopesAssignment, setActiveTab, t]);
+
+  const modalText = useMemo<{
+    title: AdminConsoleKey;
+    subtitle: AdminConsoleKey;
+    saveBtn: AdminConsoleKey;
+  }>(() => {
+    if (scopeLevel === ScopeLevel.All) {
+      return {
+        title: 'application_details.permissions.table_name',
+        subtitle: 'application_details.permissions.permissions_assignment_description',
+        saveBtn: 'general.save',
```

Comment:
I think we never used `btn` in our source code. Can you please double check and rename it to `saveButton`?

Reply:
Updated.